### ISSUE TITLE
print doesn't work in outputs function

### ISSUE THREAD

**Issue by endobson** [bazelbuild/bazel#1097]
I'm trying to debug a function that is computing the outputs for a rule. Putting a print in the implementation works, but does not work in the outputs.

My outputs function, the "here" never gets printed.

```
def _lib_outputs(attr):
  print("here")
  return {
    "zo": "%{name}.zo",
  }
```

**Comment by kchodorow** [comment]
That seems like an oversight.

**Comment by steven-johnson** [comment]
Also doesn't seem to work inside the implementation of a repository_rule().

**Comment by brandjon** [comment]
This has been fixed for output functions. Note that the prints will appear twice, because the output function is called once when the target is created (to create its predeclared outputs) and again when its rule implementation is about to be called (to populate ctx.outputs).

Not sure about repository_rule().

**Comment by laurentlb** [comment]
I think @vladmos fixed it?

**Comment by vladmos** [comment]
I must have been fixed with #076977e019ebcb8b822e7727d292636fadcafcca
